navigation-animation/src/main/java/com/google/accompanist/navigation/animation/AnimatedComposeNavigator.kt

```diff
@@ -34,7 +34,9 @@ import kotlin.collections.forEach
  */
 @ExperimentalAnimationApi
 @Navigator.Name("animatedComposable")
-public class AnimatedComposeNavigator : Navigator<AnimatedComposeNavigator.Destination>() {
+@Suppress("DEPRECATION")
+public class AnimatedComposeNavigator private constructor() : Navigator<AnimatedComposeNavigator.Destination>() {
```

Comment:
We should deprecate the class itself too, right? And it's createDestination method?

Reply:
I first went for deprecating the class but it is what prevents IDE generating replaceWith auto-complete for `Destination` inner class. That's why I went around to deprecating the constructor instead.

For `createDestination`, the IDE would auto resolve it to the correct API after refactoring `AnimatedCompostedNavigtor` and its `Destination` to androidx's version. Should we still deprecate this specifically?

Reply:
hmm let me check if references to it show as deprecated

Reply:
Nice, Class deprecation now doesn't conflict with inner class replaceWith. Added deprecation for class. Also added migration guide step `6. Replace `AnimatedComposeNavigator()` constructor with `ComposeNavigator()` constructor`

Reply:
Added deprecation to createDestination. Didn't add replaceWith because without refactoring to ComposeNavigator, replaceWith doesn't work. If navigator class was refactored, createDestination is auto-corrected to the right API.